numpy/core/src/umath/override.c

```diff
@@ -468,32 +471,34 @@ PyUFunc_CheckOverride(PyUFuncObject *ufunc, char *method,
     }
 
     len = PyTuple_GET_SIZE(normal_args);
-    override_args = PyTuple_New(len + 2);
+    override_args = PyTuple_New(len + 3);
     if (override_args == NULL) {
         goto fail;
     }
 
-    Py_INCREF(ufunc);
     /* PyTuple_SET_ITEM steals reference */
-    PyTuple_SET_ITEM(override_args, 0, (PyObject *)ufunc);
+    Py_INCREF(Py_None);
```

Comment:
Because that doesn't free the reference of the object previously in the tuple. To be honest, the reference counting design here was based on the case that I was sure I had reasoned correctly about, rather than picking a strategy that would require me to read the tuple source code again.